### ISSUE TITLE
A11y: password generator options should ideally have clearer accessible names/labels

### ISSUE THREAD

**Issue by patrickhlauke** [bitwarden/browser#1985]
## Describe the Bug

In password generator, for the option checkboxes, when announced by assistive technologies, `A-Z` and `a-z` are announced exactly the same way, `0-9` is announced as "zero nine", `!@#$%^&*` is announced as "at number dollar percent and star" (as screen readers will, by default, not announce some of those punctuation marks when presented like that as a nonsensical string in particular).

Suggest that the labels for the checkboxes here be given more explicit accessible names - visually, the form can remain as is (just showing `A-Z`, `a-z`, `0-9`, `!@#$%^&*`), but have extra text associated with the checkboxes that gives a more meaningful accessible name, such as `Uppercase (A-Z)`, `Lowercase (a-z)`, `Numbers (0-9)`, `Special characters (!@#$%^&*)`.

## Steps To Reproduce

1. Using a screen reader, navigate to the password generator
2. Using `Tab`, set focus to the various checkboxes

## Expected Result

* the checkboxes should clearly announce their purpose

## Actual Result

* `A-Z` and `a-z` are indistinguishable (announced the same way)
* `0-9` is announced as "zero nine"
* `!@#$%^&*` is announced as "at number dollar percent and star"

## Screenshots or Videos

![Password generator optional checkboxes](https://user-images.githubusercontent.com/895831/127719382-0afa44a4-cc6d-446c-8d93-000a1fe1b706.png)

## Environment

- Operating system: Windows 10
- Browser: Chrome 92
- Build Version (go to "Settings" → "About" in the app): 1.51.1

## Additional Context

Suggest changing the current markup, which (simplified) is this

```
<label for="uppercase">A-Z</label>
<input id="uppercase" type="checkbox" class="...">

<label for="lowercase">a-z</label>
<input id="lowercase" type="checkbox" class="...">

<label for="numbers">0-9</label>
<input id="numbers" type="checkbox" class="...">

<label for="special">!@#$%^&*</label>
<input id="special" type="checkbox" class="...">
```

to include explicit (longer) accessible names using `aria-label`.

```
<label for="uppercase">A-Z</label>
<input id="uppercase" type="checkbox" class="..." aria-label="Uppercase (A-Z)">

<label for="lowercase">a-z</label>
<input id="lowercase" type="checkbox" class="..." aria-label="Lowercase (a-z)">

<label for="numbers">0-9</label>
<input id="numbers" type="checkbox" class="..." aria-label="Numbers (0-9)">

<label for="special">!@#$%^&*</label>
<input id="special" type="checkbox" class="..." aria-label="Special characters (!@#$%^&*)">
```

The accessible name must still contain the visible label text (per https://www.w3.org/TR/WCAG21/#label-in-name), hence the use of the same text in brackets there. Presumably, this will require the creation of new text labels in the locales, and translation into respective languages.
